File: backend/app/schemas/prices.py
from pydantic import BaseModel

class HourlyPrice(BaseModel):
    hour: str
    timestamp_utc: str
    price_eur_mwh: float

class DayAheadPricesResponse(BaseModel):
    country: str
    zone: str
    market: str
    unit: str
    data_source: str | None = None
    prices: list[HourlyPrice]

File: frontend/types/prices.ts
export type HourlyPrice = {
    hour: string;
    timestamp_utc: string;
    price_eur_mwh: number;
};

export type DayAheadPrices ={
    country: string;
    zone: string;
    market: string;
    unit: string;
    data_source?: string;
    prices: HourlyPrice[];
};
